pkg/callback/scheduler_callback.go

```diff
@@ -139,6 +150,10 @@ func (callback *AsyncRMCallback) RecvUpdateResponse(response *si.UpdateResponse)
 				log.Logger().Error("failed to delete application", zap.Error(err))
 			}
 		} else {
+			if updated.State == events.States().Application.Killed {
+				ev := cache.NewFailApplicationEvent(updated.ApplicationID)
+				dispatcher.Dispatch(ev)
+			}
```

Comment:
When we fail an application, we need to expose some pod level events to indicate this issue.
we can do this in a follow up JIRA.